### ISSUE TITLE
Any way to dynamically change the schema?

### ISSUE THREAD

**Issue by sithwin** [makinacorpus/angular2-schema-form#39]
I have dropdown list and would like to change Json Schema when it is fired the change event. 

Pls advice!

**Comment by ebrehault** [comment]
Sure, the schema is just an attribute of the form component, you can change it whenever you want.

**Comment by sithwin** [comment]
Currently, it is not dynamically changed.

Here is my dropdown list to load all the schema:
`<select [(ngModel)]="selectedValue" class="form-control" name="contentType" id="contentType" (ngModelChange)="updateContent($event)" *ngIf="!isUpdate">
        <option *ngFor="let contentType of contentTypeEntities" [ngValue]="contentType" >
          {{contentType.name}}
        </option>
      </select>`

Here is my schema form table:
`<table ngClass="jsonTable">
        <tr>
          <td>
            <schema-form id="schemaForm" [model]="cmsContent.contentJson" [schema]="contentType.jsonSchema" (onChange)="value=($event.value)">
            </schema-form>
          </td>
        </tr>`

Here is ts:
`updateContent(selectedContenType: ContentTypeDto) {
    this.contentType = selectedContenType;
  }`

I would like to dynamically change the schema form when the dropdown list is fired change event.

Pls advice what I am doing wrong. Thanks!

**Comment by sithwin** [comment]
Currently, it is not dynamically changed.

Here is my dropdown list to load all the schema:
`<select [(ngModel)]="selectedValue" class="form-control" name="contentType" id="contentType" (ngModelChange)="updateContent($event)" *ngIf="!isUpdate">
        <option *ngFor="let contentType of contentTypeEntities" [ngValue]="contentType" >
          {{contentType.name}}
        </option>
      </select>`

Here is my schema form table:
`      <table ngClass="jsonTable">
        <tr>
          <td>
            <schema-form id="schemaForm" [model]="cmsContent.contentJson" [schema]="contentType.jsonSchema" (onChange)="value=($event.value)">
            </schema-form>
          </td>
        </tr>
      </table>`

Here is ts:
`updateContent(selectedContenType: ContentTypeDto) {
    this.contentType = selectedContenType;
  }`

I would like to dynamically change the schema form when the dropdown list is fired change event.

Pls advice what I am doing wrong. Thanks!

**Comment by kianmeng-aft** [comment]
Hey , i am facing the same issue too.

Steps to reproduce,

1) Clone the demo out `https://github.com/fbessou/angular2-schema-form-demo`
2) Duplicate `sampleschema.json` to `sampleschema.1.json` and remove some fields
3) Add a new button`	<button (click)="changeSchema()"> Change schema </button>` in `app.component.html`
4) Add new method to change schema in `app.component.ts`
```
changeSchema(){
		this.schema = require("./sampleschema.1.json");
	}
```

The final json generated on the right side, seems to change when the new button is clicked... but the UI does not change (Which is what i want)

Is it suppose to work this way? Because , i will have a couple of forms, and i want it to change it dynamically.

Thanks in advance!

**Comment by eesdil** [comment]
The change detection does not fire or the reference is not changing for the formProperty on the directive...
Any idea where is the issue?

**Comment by kianmeng-aft** [comment]
any update on this?

**Comment by anna2102** [comment]
Hi,

I am looking for the same solution to render UI immediately as soon as schema-form is updated? Is there any ways to do so? Thanks.

**Comment by SBats** [comment]
Hi,
We have indeed an issue with changes propagation.
At the moment we're working on fixing AOT compatibility but this will be our next concern.
Sorry for the inconvenience!

**Comment by YoYo35** [comment]
Hi
Any estimation for this issue? I am also facing the same issue.

**Comment by YoYo35** [comment]
Or can I use early release to fix this problem?

**Comment by levequej** [comment]
same issue here (in my case I want to achieve what is mentionnent in #49) , even the example in https://github.com/makinacorpus/angular2-schema-form#fixing-the-schema-or-model-before-rendering does not work for me with angular 2.3.1 using angular-cli":  1.0.0-beta.26 and angular2-schema-form 1.0.0-alpha.33 .  It seems to worknig only when providing a static JSON

**Comment by ruben0909** [comment]
+1

**Comment by fbessou** [comment]
Fixed by #86
